gridcapa-cse-valid-app/src/main/java/com/farao_community/farao/cse_valid/app/ZonalScalableProvider.java

```diff
@@ -0,0 +1,111 @@
+ */
+
+package com.farao_community.farao.cse_valid.app;
+
+import com.farao_community.farao.cse_valid.api.exception.CseValidInvalidDataException;
+import com.farao_community.farao.cse_valid.api.resource.ProcessType;
+import com.farao_community.farao.cse_valid.app.configuration.EicCodesConfiguration;
+import com.powsybl.glsk.commons.ZonalData;
+import com.powsybl.iidm.modification.scalable.Scalable;
+import com.powsybl.iidm.network.Country;
+import com.powsybl.iidm.network.Load;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.Substation;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Vincent BOCHET {@literal <vincent.bochet at rte-france.com>}
+ */
+@Service
+public class ZonalScalableProvider {
+
+    private final FileImporter fileImporter;
+    private final EicCodesConfiguration eicCodesConfiguration;
+
+    public ZonalScalableProvider(FileImporter fileImporter, EicCodesConfiguration eicCodesConfiguration) {
+        this.fileImporter = fileImporter;
+        this.eicCodesConfiguration = eicCodesConfiguration;
+    }
+
+    public ZonalData<Scalable> get(String glskUrl, Network network, ProcessType processType) {
+        ZonalData<Scalable> zonalScalable = fileImporter.importGlsk(glskUrl).getZonalScalable(network);
+        eicCodesConfiguration.getCseCodes().forEach(eic -> checkCseCountryInGlsk(zonalScalable, eic));
+        stackScalableOnLoads(network, zonalScalable, processType);
+        return zonalScalable;
+    }
+
+    private static void checkCseCountryInGlsk(ZonalData<Scalable> zonalScalable, String eic) {
+        if (!zonalScalable.getDataPerZone().containsKey(eic)) {
+            throw new CseValidInvalidDataException(String.format("Area '%s' was not found in the glsk file.", eic));
+        }
+    }
+
+    private void stackScalableOnLoads(Network network, ZonalData<Scalable> zonalScalable, ProcessType processType) {
+        zonalScalable.getDataPerZone().forEach((eiCode, scalable) -> {
+            if (processType == ProcessType.IDCC && eiCode.equals(eicCodesConfiguration.getItaly())) {
+                return;
+            }
+            double sum = getZoneSumOfActiveLoads(network, eiCode);
+            // No need to go further if a country has no active load
+            if (sum == 0.0) {
+                return;
+            }
+            Scalable stackedScalable = getStackedScalable(eiCode, scalable, network, sum);
+            zonalScalable.getDataPerZone().put(eiCode, stackedScalable);
+        });
+    }
+
+    private double getZoneSumOfActiveLoads(Network network, String cseCountry) {
+        return network.getLoadStream()
+            .filter(load -> isLoadCorrespondingToTheCountry(load, cseCountry))
+            .map(Load::getP0)
+            .reduce(0., Double::sum);
+    }
+
+    private Scalable getStackedScalable(String cseCountry, Scalable scalable, Network network, double sum) {
```

Comment:
static and rename cseCountry to eiCode

Reply:
Same : Renaming OK but method can't be static.